ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java

```diff
@@ -131,11 +130,17 @@ public CompletableFuture<FastSyncState> downloadPivotBlockHeader(
   private CompletableFuture<FastSyncState> internalDownloadPivotBlockHeader(
       final FastSyncState currentState) {
     if (currentState.hasPivotBlockHeader()) {
-      LOG.info("Initial sync state {} already contains the block header", currentState);
+      LOG.debug("Initial sync state {} already contains the block header", currentState);
       return completedFuture(currentState);
     }
 
-    return waitForPeers(1)
+    return ethContext
+        .getEthPeers()
+        .waitForPeer(
+            (peer) ->
+                currentState.getPivotBlockNumber().isEmpty()
```

Comment:
Why is this checking for a missing pivot block number? The if statement at the top of this function ensures we only wait for a peer if we don't already have the pivot block header.

Reply:
Initially, I thought this too, but iirc when I tried to run the code, it would result in exceptions from the pivot block number being unset  but having getAsLong() called